composer.json

```diff
@@ -39,11 +39,11 @@
         "linkorb/jsmin-php": "~1.0",
         "wikimedia/less.php": "~3.0",
         "scssphp/scssphp": "~1.0",
-        "symfony/yaml": "^3.4",
+        "symfony/yaml": "^5.1",
```

Comment:
I have not looked into changes, I have no overview over breaking changes. Are there unit tests that adress stuff that break? It passed all current unit tests in storm? I assumed the current unit tests address all current stress points in the library? If you have an overview of bug reports of yaml that breaks I can take a targetted look.

Reply:
The breaking changes from 3 to 4 means that the usual version.yaml in use fails:

```yaml
1.1.2: !!! Important update
```
needs to be 
```yaml
"1.1.2": "!!! Important update"
```
in YAML 4. I haven't looked at any changes from 4 to 5 though.

Reply:
Hmmh what i've found is that when a string values starts with an exclamation mark `https://github.com/symfony/yaml/blob/afe64a9ce7718524116c9a3b0913e5163b49c405/Inline.php#L585` it will check if it matches a scalar, and that will break it. 

Valid scalars are 

```php
$valid_scalar_tags = [
                      '!!str ',
                      '! ',
                      '!php/object',
                      '!php/const',
                      '!!float ',
                      '!!binary '
];
```
So the string should be wrapped with quotes when it doesn't match a valid scalar, something in the line of

```php
public function checkValue($text)
{
  if(strpos($text, '!') === 0) {
    $valid_scalar_tags = [
                          '!!str ',
                          '! ',
                          '!php/object',
                          '!php/const',
                          '!!float ',
                          '!!binary '
    ];
    $lower = strtolower($text);
    foreach($valid_scalar_tags as $scalar) {
      if(strpos(lower, $scalar) === 0) {
        return $text;
      }
    }   
  return '"'.$text.'"';
  }
return $text;
}
```

or put a `! ` in front of it so it just gets substringed out and not considered as a scalar, so you don't have to worry about quotes https://github.com/symfony/yaml/blob/afe64a9ce7718524116c9a3b0913e5163b49c405/Inline.php#L589

I'm not sure which class handles the version.yalm loading, and I honestly don't have the time to write a pre processor or specialised class for this use case where a scalar interpretation is attempted.

I added a version.yaml test in the unit tests, and that works properly, the keys don't need to be quoted in 5.1 to be interpreted correctly as far as I can see, so it's only the scalar tags that cause issue in your provided example.

Reply:
We wouldn't want the version.yaml pre-processing to be done in the core anyways, see https://github.com/wintercms/winter/commit/58664605ea53465422882a1b543bc9d7a32b755c for how it's handled in the system module instead.

Reply:
Do you want me to remove the version.yaml unit test then from this library?

Reply:
That's probably fine to do yeah

Reply:
Done